s3tests_boto3/functional/test_s3.py

```diff
@@ -1748,6 +1783,48 @@ def test_bucket_list_return_data_versioning():
         eq(obj['VersionId'], key_data['VersionId'])
         _compare_dates(obj['LastModified'],key_data['LastModified'])
 
+@attr(resource='object')
+@attr(method='head')
+@attr(operation='compare w/bucket list when bucket versioning is configured after object acl changed')
+@attr(assertion='return same metadata')
+@attr('versioning')
+def test_bucket_list_return_data_versioning_after_object_acl_changed():
+    bucket_name = get_new_bucket()
+    check_configure_versioning_retry(bucket_name, "Enabled", "Enabled")
+    key_names = ['bar', 'baz', 'foo']
+    bucket_name = _create_objects(bucket_name=bucket_name,keys=key_names)
+
+    client = get_client()
+    data = {}
+
+    for key_name in key_names:
+        client.put_object_acl(ACL='public-read',Bucket=bucket_name, Key=key_name)
+        obj_response = client.head_object(Bucket=bucket_name, Key=key_name)
+        acl_response = client.get_object_acl(Bucket=bucket_name, Key=key_name)
+        data.update({
+            key_name: {
+                'ID': acl_response['Owner']['ID'],
+                'DisplayName': acl_response['Owner']['DisplayName'],
+                'ETag': obj_response['ETag'],
+                'LastModified': obj_response['LastModified'],
+                'ContentLength': obj_response['ContentLength'],
+                'VersionId': obj_response['VersionId']
+                }
+            })
+
+    response  = client.list_object_versions(Bucket=bucket_name)
+    objs_list = response['Versions']
+
+    for obj in objs_list:
+        key_name = obj['Key']
+        key_data = data[key_name]
+        eq(obj['Owner']['DisplayName'],key_data['DisplayName'])
+        eq(obj['ETag'],key_data['ETag'])
+        eq(obj['Size'],key_data['ContentLength'])
+        eq(obj['Owner']['ID'],key_data['ID'])
+        eq(obj['VersionId'], key_data['VersionId'])
+        _compare_dates(obj['LastModified'],key_data['LastModified'])
```

Comment:
running this test against ceph's master branch (without the fix from https://github.com/ceph/ceph/pull/42191), i'm seeing this test pass most of the time. rarely i see it fail with
> AssertionError: datetime.datetime(2021, 8, 3, 18, 37, 50, tzinfo=tzutc()) != datetime.datetime(2021, 8, 3, 18, 37, 51, tzinfo=tzutc())

adding a `sleep(2)` after `_create_objects()` allowed the test to fail consistently